runtime/src/wdt.rs

```diff
@@ -0,0 +1,105 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    wdt.rs
+
+Abstract:
+
+    File contains execution routines for programming the Watchdog Timer.
+
+
+--*/
+
+use caliptra_drivers::SocIfc;
+
+use crate::cprintln;
+use core::cmp::max;
+
+pub struct WdtTimeout(pub core::num::NonZeroU32);
+
+impl Default for WdtTimeout {
+    fn default() -> WdtTimeout {
+        WdtTimeout::new(WDT1_DEFAULT_TIMEOUT_SECS)
+    }
+}
+
+impl WdtTimeout {
+    pub fn new(timeout_secs: u32) -> Self {
+        match core::num::NonZeroU32::new(timeout_secs) {
+            Some(val) => Self(val),
+            None => panic!("WdtTimeout cannot be 0"),
+        }
+    }
+}
+
+impl From<core::num::NonZeroU32> for WdtTimeout {
+    fn from(val: core::num::NonZeroU32) -> Self {
+        WdtTimeout(val)
+    }
+}
+impl From<WdtTimeout> for core::num::NonZeroU32 {
+    fn from(val: WdtTimeout) -> Self {
+        val.0
+    }
+}
+impl From<WdtTimeout> for u32 {
+    fn from(val: WdtTimeout) -> Self {
+        core::num::NonZeroU32::from(val).get()
+    }
+}
+
+const EXPECTED_CALIPTRA_BOOT_TIME_IN_CYCLES: u64 = 20_000_000; // 20 million cycles
+const WDT1_DEFAULT_TIMEOUT_SECS: u32 = 5;
```

Comment:
I heard we want to use cycle counts instead of relying on second -> cycle conversions. Especially since we want to be sure we have the same behavior on different emulation platforms.

Should we remove the wall-clock based stuff?

Reply:
Done.